### PR TITLE
[SPARK-45973][INFRA] re-org `apt-get` installations

### PR DESCRIPTION
**Pull request by zhengruifeng**
### What changes were proposed in this pull request?
`apt-get` installation clean up, after this PR, there is no `apt-get` command in `build_and_test`

### Why are the changes needed?
most `apt` packages (except `ruby`/`ruby-dev`) in `build_and_test` were already installed in dockerfile, seems no need to reinstall

### Does this PR introduce _any_ user-facing change?
no

### How was this patch tested?
ci

### Was this patch authored or co-authored using generative AI tooling?
no

### CODE REVIEW

**Comment by zhengruifeng** [comment]
`Docker integration tests` failure seems unrelated, since it is failing in master

**Comment by zhengruifeng** [comment]
cc @dongjoon-hyun @HyukjinKwon @Yikun

File: dev/infra/Dockerfile
Comment (dongjoon-hyun) [inline dev/infra/Dockerfile:61]: Is this sorted simply, @zhengruifeng ?

**Review by dongjoon-hyun — APPROVED** [review]
+1, LGTM.

File: dev/infra/Dockerfile
Comment (zhengruifeng) [reply dev/infra/Dockerfile:61]: yes, thanks for the reviews

File: .github/workflows/build_and_test.yml
Comment (LuciferYang) [inline .github/workflows/build_and_test.yml:715]: After this pr merged, the daily tests of branch-3.x started to lack dependencies ...should similar optimizations all seem to be backported?

-branch-3.3 https://github.com/apache/spark/actions/runs/6918980555/job/18821819222
-branch-3.4 https://github.com/apache/spark/actions/runs/6919512513/job/18822994877
-branch-3.5 https://github.com/apache/spark/actions/runs/6920069263/job/18824167689

image

or should disable lint check for branch-3.x daily tests?

Comment (dongjoon-hyun) [reply .github/workflows/build_and_test.yml:715]: Oh, too bad. I don't prefer backporing the infra PR, but I'll leave this to @zhengruifeng .

Comment (zhengruifeng) [reply .github/workflows/build_and_test.yml:715]: oops, let me take a look

Comment (zhengruifeng) [reply .github/workflows/build_and_test.yml:715]: it is hard to backport, I am going to restore installations in https://github.com/apache/spark/pull/43904

**Pull request closed without merging**
